### ISSUE TITLE
Crosshairs broken in v1.5.4

### ISSUE THREAD

**Issue by matoos32** [jfree/jfreechart#340]
Going from release v1.5.3 to v1.5.4 crosshair functionality stoped working. Clicking on charts no longer updates the crosshair.

To reproduce:

*Setup*
1. Find demos with crosshairs: https://github.com/jfree/jfree-demos/search?q=crosshair
**Note:** jfree-demos currently [lists JfreeChart v1.5.3 in the README](https://github.com/jfree/jfree-demos/blob/master/README.md?plain=1#L18) but doesn't have a v1.5.3 release tag. It's up to v1.5.2.
1. `git clone https://github.com/jfree/jfree-demos.git`
1. `cd jfree-demos`
1. `git checkout -b crosshair-test v2.0`

*Test v1.5.2*
1. `mvn clean package`
1. `java -cp target/org.jfree.demos-2.0-jar-with-dependencies.jar org.jfree.chart.demo2.TimeSeriesChartDemo1`
1. Observe clicking the chart updates the crosshair.
1. Close the app

Test v1.5.3
1. Set the POM jfreechart [dependency](https://github.com/jfree/jfree-demos/blob/v2.0/pom.xml#L50) to v1.5.**3** 
1. Repeat build and launch steps.
1. Observe clicking the chart updates the crosshair.

Test v1.5.4
1. Set the POM jfreechart [dependency](https://github.com/jfree/jfree-demos/blob/v2.0/pom.xml#L50) to v1.5.**4**
1. Repeat build and launch steps.
1. Observe the crosshair doesn't update. If you resize the window (trigger a re-draw) the crosshair updates to the last clicked location.

I suspect it's related to a [v1.5.4 contribution](https://github.com/jfree/jfreechart/releases/tag/v1.5.4).

**Comment by trashgod** [comment]
For reference, I can reproduce the anomaly in `TimeSeriesChartDemo1` and similar examples that make the crosshairs visible on the plot, but not in examples that use `CrosshairOverlay`, seen here(https://stackoverflow.com/a/21180275/230513) and [here](https://gist.github.com/trashgod/a3f7581417db19bf5b6dc52d3647a2f9).

**Comment by jfree** [comment]
At first glance it seems to be caused by the removal of this line in `ChartPanel.java` between v1.5.3 and v1.5.4:

<img width="865" alt="image" src="https://user-images.githubusercontent.com/1835893/212496339-9eeb59ef-8f9b-43fd-8eb3-c71beaa14e93.png">

**Comment by matoos32** [comment]
Thanks for having taken a look!

**Comment by trashgod** [comment]
Yes, thank you. After restoring the call to `setNotify()`, I can no longer reproduce the anomaly. Related discussion may be found [here](https://stackoverflow.com/search?q=%5Bjfreechart%5D+setNotify) and [here](https://www.jfree.org/forum/viewtopic.php?f=3&t=116135&p=175587&hilit=setNotify#p175587).

**Comment by Yuri-Blankenstein-TNO** [comment]
> At first glance it seems to be caused by the removal of this line in `ChartPanel.java` between v1.5.3 and v1.5.4:
> 
> <img alt="image" width="865" src="https://user-images.githubusercontent.com/1835893/212496339-9eeb59ef-8f9b-43fd-8eb3-c71beaa14e93.png">

I know exactly when this was introduced, see https://github.com/jfree/jfreechart/pull/233#discussion_r655296139.
The problem now is that large charts are re-rendered every time when a user clicks with the mouse.
This leads to very laggy behavior when selecting chart elements, though the selection itself is rendered using an ChartOverlay which is very fast.

**Comment by Yuri-Blankenstein-TNO** [comment]
I've created an alternative solution that prevents charts without corsshairs to redraw on every mouse click, see #362.

For this I have reused - what seems - an older implementation [JFreeChart#handleClick(...)](https://github.com/jfree/jfreechart/blob/b9a36f260cd243ddea6cbfd603f6db32dcf7565f/src/main/java/org/jfree/chart/JFreeChart.java#L1378) that exactly seems to solve this issue. This public method isn't called anywhere else anymore within the library.
